tests/integration/test_create_token.py

```diff
@@ -0,0 +1,203 @@
+
+EXPECTED_TOKEN_BUSINESS = {
+    "account_service_url": "http://upstream.url",
+    "case_id": "1001",
+    "collection_exercise_sid": "789",
+    "exp": 1709058195.091798,
+    "iat": 1709054595.091798,
+    "jti": "1001",
+    "language_code": "en",
+    "response_expires_at": "2024-02-28T09:59:43.109276+00:00",
+    "response_id": "1234567890123456",
+    "roles": [],
+    "schema_name": "test_metadata_routing.json",
+    "survey_metadata": {
+        "data": {
+            "display_address": "68 Abingdon Road, Goathill",
+            "employment_date": "1983-06-02",
+            "flag_1": 123,
+            "link": "https://example.com",
+            "period_id": "abc",
+            "period_str": "April 2016",
+            "ref_p_end_date": "2016-04-30",
+            "ref_p_start_date": "2016-04-01",
+            "ru_name": "Integration Testing",
+            "ru_ref": "12345678901A",
+            "trad_as": "Integration Tests",
+            "user_id": "integration-test",
+        }
+    },
+    "tx_id": "1001",
+    "version": "v2",
+}
+EXPECTED_TOKEN_SOCIAL = {
+    "account_service_url": "http://upstream.url",
+    "case_id": "1001",
+    "collection_exercise_sid": "789",
+    "exp": 1709058195.091798,
+    "iat": 1709054595.091798,
+    "jti": "1001",
+    "language_code": "en",
+    "response_expires_at": "2024-02-28T09:59:43.109276+00:00",
+    "response_id": "1234567890123456",
+    "roles": [],
+    "schema_name": "social_demo.json",
+    "survey_metadata": {
+        "data": {
+            "case_ref": "1000000000000001",
+            "date": "2016-05-12",
+            "flag_1": True,
+            "qid": PAYLOAD_V2_SOCIAL.get("survey_metadata").get("data").get("qid"),
+            "user_id": "abc",
+        },
+        "receipting_keys": ["qid"],
+    },
+    "tx_id": "1001",
+    "version": "v2",
+}
+
+
+class TestCreateToken(IntegrationTestCase, AppContextTestCase):
+    """
+    The purpose of this test class is to test the creation of a token (from create_token.py) to ensure
+    metadata in a decrypted token is nested/found in the correct level.
+    """
+
+    # Patches are used since the values "uuid4", "time" and "get_response_expires_at" return are dynamic
+    @mock.patch("tests.integration.create_token.uuid4")
+    @mock.patch(
+        "tests.integration.create_token.time",
+    )
+    @mock.patch(
+        "tests.integration.create_token.get_response_expires_at",
+    )
+    def test_payload_content_and_structure_from_token(
+        self, mock_response_expiry_time, mock_time, mock_uuid
+    ):
+        mock_uuid.return_value = 1001
+        mock_time.return_value = 1709054595.091798
+        mock_response_expiry_time.return_value = "2024-02-28T09:59:43.109276+00:00"
+        test_parameters = [
+            {
+                "schema": "test_metadata_routing.json",
+                "theme": "default",
+                "additional_payload": {
+                    "flag_1": 123,
+                    "period_id": "abc",
```

Comment:
Minor but I would update the period_id to the kind of period_id we would expect e..g YYYYMM format

Reply:
Makes sense, updated 👍